opennms-provision/opennms-detector-jms/src/main/java/org/opennms/netmgt/provision/detector/jms/ActiveMQDetector.java

```diff
@@ -0,0 +1,161 @@
+
+import java.lang.reflect.UndeclaredThrowableException;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import javax.jms.Connection;
+import javax.jms.Session;
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.opennms.netmgt.provision.DetectRequest;
+import org.opennms.netmgt.provision.support.AgentBasedSyncAbstractDetector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>ActiveMQDetector class.</p>
+ *
+ * @author roskens
+ * @version $Id: $
+ */
+
+public class ActiveMQDetector extends AgentBasedSyncAbstractDetector<URI>{
+    private static final Logger LOG = LoggerFactory.getLogger(ActiveMQDetector.class);
+
+    private static final String DEFAULT_SERVICE_NAME = "ActiveMQ";
+    private static final String DEFAULT_BROKERURL = "vm://localhost?create=false";
+
+    private String m_brokerURL;
+    private String m_user;
+    private String m_password;
+    private Boolean m_useNodeLabel;
+    private final ActiveMQConnectionFactory m_connectionFactory;
+    private Connection connection;
+    private Session session;
+
+    /**
+     * <p>Constructor for ActiveMQDetector.</p>
+     */
+    public ActiveMQDetector() {
+        super(DEFAULT_SERVICE_NAME, 0);
+        setBrokerURL(DEFAULT_BROKERURL);
+        m_connectionFactory = new ActiveMQConnectionFactory();
+        connection = null;
+        session = null;
+    }
+
+    public ActiveMQDetector(String serviceName, int port) {
+        super(serviceName, port);
+        setBrokerURL(DEFAULT_BROKERURL);
+        m_connectionFactory = new ActiveMQConnectionFactory();
+        connection = null;
+        session = null;
+    }
+
+    @Override
+    public URI getAgentConfig(DetectRequest request) {
+        URI uri = null;
+        try {
+            uri = new URI(m_brokerURL);
+            Map<String,String> map = request.getRuntimeAttributes();
+            if (map != null) {
+                LOG.info("runtimeAttributes: {}", map);
+                //uri = new URI(uri.getScheme(), uri.getUserInfo(), map.getNodeLabel(), uri.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
+            }
+        } catch (URISyntaxException e) {
+            LOG.error("Invalid BrokerURL: {}", m_brokerURL, e);
+        }
+        return uri;
+    }
+
+    @Override
+    public boolean isServiceDetected(InetAddress address, URI agentConfig) {
+        LOG.debug("isServiceDetected(address=[{}], agentConfig=[{}])", address, agentConfig);
+        try {
+            m_connectionFactory.setBrokerURL(agentConfig.toString());
+            connection = m_connectionFactory.createConnection(m_user, m_password);
+            LOG.info("connection created");
+            session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            LOG.info("session created");
+            connection.start();
+            LOG.info("connection started");
+
+            session.close();
```

Comment:
I've moved the closing of the connection and session objects into the dispose() method.